runagent/sdk/template_manager.py

```
@@ -21,11 +22,20 @@ def __init__(self):
         )
 
     def check_connectivity(self) -> bool:
-        """Check if template repository is accessible"""
+        """Check if template repository is accessible using git ls-remote (lightweight approach)"""
         try:
-            self.list_available()
-            return True
-        except Exception:
+            # Use git ls-remote to check repository accessibility without cloning
+            # This is much faster than the previous approach of calling list_available()
+            result = subprocess.run(
```

Ruff (0.13.1)

`subprocess` call: check for execution of untrusted input

(S603)